src/ui/RadarMarker.ts

```diff
@@ -97,33 +125,64 @@ class RadarMarker extends maplibregl.Marker {
 
       const onError = (err: any) => {
         Logger.error(`Could not load marker: ${err.message} - falling back to default marker`);
+        IMAGE_CACHE.set(markerOptions.url as string, 'failed'); // mark as failed
         this._element.replaceChildren(...Array.from(originalElement.childNodes));
       }
 
+      // custom URL image
       if (markerOptions.url) {
-        fetch(markerOptions.url)
-          .then(res => {
-            if (res.status === 200) {
-              res.blob()
-                .then(onSuccess)
-                .catch(onError);
-            } else {
-              onError(new Error(res.statusText));
+        const loadImage = () => { // fetch marker data from URL
+          fetch(markerOptions.url as string)
+            .then(res => {
+              if (res.status === 200) {
+                res.blob()
+                  .then((data) => {
+                    IMAGE_CACHE.set(markerOptions.url as string, data); // cache data
+                    onSuccess(data);
+                  })
+                  .catch(onError);
+              } else {
+                onError(new Error(res.statusText));
+              }
+            })
+            .catch(onError)
+        };
+
+        // attempt to use cached data, otherwise fetch marker image data from URL
+        useCachedImage(markerOptions.url)
+          .then(onSuccess)
+          .catch((reason: 'miss' | 'timedout' | 'failed' | Error) => {
+            if (reason !== 'miss') {
+              Logger.debug(`RadarMarker: cache lookup for ${markerOptions.url}: ${reason}`);
             }
-          })
-          .catch(onError)
+            loadImage();
```

Comment:
might be nice to just have `useCachedImage` do the 1) check if in cache 2) if its not, load it

Reply:
The only think that makes it sorta tricky is the image loading happens differently depending on the type of marker being loaded (either from a remote URL, or via authenticated request to Radar). Can maybe play around with it.

Reply:
yea maybe could pass in a `loadImage` function, but not a huge deal